frontend/src/components/Ingredients/Ingredients.tsx

```diff
@@ -1,19 +1,91 @@
-import { useEffect, useState } from 'react'
-import { fetchAllIngredients } from '../apiCalls'
-import './Ingredients.css'
+import { Ingredient } from '../apiCalls';
 
+interface Props {
+  ingredients: Array<Ingredient>;
+  setIngredients: Function;
+}
 
-export function Ingredients() {
+export function Ingredients({ ingredients, setIngredients }: Props) {
 
-  const [ingredients, setIngredients] = useState([])
-  
-  useEffect(() => {
-    fetchAllIngredients().then(data => setIngredients(data))
-  }, [])
+  function sortIngredients(criteria: keyof Ingredient) {
+    const sortedIngredients = [...ingredients].sort((a, b) => {
+      const aAttribute = a[criteria]
+      const bAttribute = b[criteria]
+      if(typeof aAttribute === "number" && typeof bAttribute === "number") {
+        return aAttribute - bAttribute
+      }
+      return (aAttribute as string).localeCompare(bAttribute as string)
+    });
+    setIngredients(sortedIngredients)
+  }
+
+  const renderedIngredients = ingredients.map((ingredient) => {
+    return (
+      <tr key={ingredient.id}>
```

Comment:
I'd like to see this whole `<tr>` as a separate `<Ingredient ... />` component (then move the `{ingredients.map(ingredient => <Ingredient ... />)}` into the main JSX since it'll be tidier).

Reply:
Can do. Agreed.